crates/daemon/src/monitor.rs

```diff
@@ -466,25 +470,76 @@ impl Actor {
 
         tracing::trace!("Updating status of {num_transactions} transactions",);
 
-        let histories = self
-            .client
-            .batch_script_get_history(self.state.monitoring_scripts())
-            .context("Failed to get script histories")?;
-
-        let mut ready_events = self.state.update(
-            latest_block_height,
-            histories
-                .into_iter()
-                .map(|list| {
-                    list.into_iter()
-                        .map(|response| TxStatus {
-                            height: response.height,
-                            tx_hash: response.tx_hash,
-                        })
-                        .collect()
-                })
-                .collect(),
-        );
+        // TODO: Evaluate buffer; if we do batches of 25 and allow buffer of 100 here, does that
+        // mean we can at most process 4 batches at the time?  What happens if the buffer is
+        // exhausted? (I would expect the next messages will be queued) We use tokio::sync::
+        // mpsc::channel because we need the receiver to be in an async context because we do async
+        // work upon receiving script updates
+        let (sender, mut receiver) = tokio::sync::mpsc::channel(100);
+
+        let scripts = self
+            .state
+            .monitoring_scripts()
+            .cloned()
+            .collect::<Vec<Script>>();
+
+        let batches = scripts.chunks(BATCH_SIZE).map(|batch| batch.to_owned());
+
+        let client = self.client.clone();
+
+        let batch_futures = batches.into_iter().map(move |batch| {
```

Comment:
🔧
I think you will need to wait between the batches, otherwise you just fire one after the other because you immediately spawn them away.

Reply:
It would be interesting to see whether we actually need to wait here; we're not concerned about the results coming out of order, as long as the results will come back (depends on the behaviour of Electrum?)

Reply:
If we don't wait here, we probably don't have to use batches because they get all spawned away.

Reply:
well, the difference is still that we're partitioning them in different worker threads (the `spawn_blocking`)

Reply:
So we will spawn batches of 25 in parallel and execute all (4x) 25 in parallel and wait until at least 1 is finished before we spawn another 25?

Reply:
pretty much, but the 4x figure depends on the limit of the worker threads. AFAICT, this limit is by default 512, so we pretty much fire all the batches at the same time, and then reach the electrum API one at a time from all these threads.
tokio has separate threads, separate from the runtime that are used here.

Reply:
gotcha, thanks for the explanation. :)

Reply:
Maybe I misunderstand what you are saying.

The idea of this implementation is to chunk up the requests in packages of 25 and then process all these packages (I called them batches) in parallel.
Maybe the term `batch` is not used correctly here, because we are not processing what is in the batch in parallel, but we process all the chunks in parallel. 
I am happy to rename what is called `batch` at the moment. 
Do you think the implementation needs changes?

---

Note that in fixup https://github.com/itchysats/itchysats/pull/3190/commits/c009237fd04b7258b066e4d867323d77fd5d6785 we stop joining all the `spawn_blocking` join handles, because that is actually necessary and just adds complexity.